weaver/src/main/java/org/aspectj/weaver/reflect/Java15AnnotationFinder.java

```diff
@@ -41,9 +41,17 @@
 
 	private Repository bcelRepository;
 	private BcelWeakClassLoaderReference classLoaderRef;
+
+	private static Repository staticBcelRepository;
+	private static BcelWeakClassLoaderReference staticClassLoaderRef;
+
 	private World world;
 	private static boolean useCachingClassLoaderRepository;
-	
+
+	//Use single instance of Repository and ClassLoader
+	public static final boolean useSingleInstances =
+		System.getProperty("org.aspectj.apache.bcel.useSingleRepositoryInstance", "true").equalsIgnoreCase("true");
```

Comment:
This automatically activates the optimisations. Depending on what @aclement wants, the defaults for this property would have to be `"false"` (or anything else than `"true"`) for backward compatibility. Only if we decide to auto-activate the optimisations, this can stay like it is now.